addons/sale_project/models/product.py

```diff
@@ -15,7 +15,10 @@ def _selection_service_policy(self):
             ('ordered_prepaid', _('Prepaid/Fixed Price')),
             ('delivered_manual', _('Based on Delivered Quantity (Manual)')),
         ]
-        if self.user_has_groups('project.group_project_milestone'):
+        user = self.env['res.users'].sudo().browse(SUPERUSER_ID)
+        if (self.user_has_groups('project.group_project_milestone') or
+                (self.env.user.has_group('base.group_public') and user.has_group('project.group_project_milestone'))
```

Comment:
We could also have the same issue with a portal user, no? If yes, then we could instead check if the user is external one.

```suggestion
                (self.env.user.share and user.has_group('project.group_project_milestone'))
```

What do you think? 🙂

Reply:
I have actually tested it and portal user doesn't face the same issue it just happen when you are a public user and you are not signed in. :slightly_smiling_face:

Reply:
Ah indeed, it is because we also give that group for the portal user. 🙂 

https://github.com/odoo/odoo/blob/7e3403068fc3fbc40182b3cfeb80e97a9300e8ff/addons/project/models/res_config_settings.py#L17

Thanks for your testing! 🙂